File: MovieClient.java
// KEVIN ANTONIO CHÁVEZ LARA - 185380
import java.io.*;
import java.net.*;

class MovieClient {

	public static void main(String[] args) throws IOException
	{
        // CREAR LA OPCION DE ARGUMENTOS PARA HOOTNAME Y PORT
        if (args.length < 2)
        { 
            System.out.println("Modo de uso: MovieClient IPAddress portNumber");
            System.exit(1);
        }
        int port = Integer.parseInt(args[1]); 
        String hostName = args[0]; 
        // Open your connection to a server, at port 1234
        Socket s1 = new Socket(hostName,port);
		// Cliente 
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Nombre de la pelicula a buscar: ");
        String nombrePelicula = in.readLine(); // Lee el nombre de la pelicula.
        System.out.print("\nOpciones de busqueda:\n- Genre            1\n- Lead Studio      2\n- Audience score   3\n- Profitability    4\n- Rotten Tomatoes  5\n- Worldwide Gross  6\n- Year             7\nElija la opcion: ");
        String opcionConsulta = in.readLine(); // Lee la opcion/opcode.
        // CREAMOS EL MENSAJE DE PROTOCOLO CON OPCODE + STRING
        String mensajeProtocolo = new String(opcionConsulta);
        mensajeProtocolo = mensajeProtocolo + "," + nombrePelicula;
		// Mensaje del protocolo se envia al server utilizando un socket:
        OutputStream s1out = s1.getOutputStream();
        DataOutputStream dos = new DataOutputStream (s1out);
        dos.write(mensajeProtocolo.getBytes("utf-8"));
		// Server recibe el mensaje y lo procesa, para luego mandar el resultado:

        InputStream s1In = s1.getInputStream();
        DataInputStream dis = new DataInputStream(s1In);

        byte[] buffer = new byte[255];
        dis.read(buffer);
        String resultadoProtocolo= new String (buffer);
        String[] mensajeSplit = resultadoProtocolo.split(",");

        System.out.println("El resultado de la busqueda fue: ");
        if (Integer.parseInt(mensajeSplit[0]) == 1)
        {
            System.out.print("EXITOSO: ");
            System.out.println(mensajeSplit[1]);
        }
        else
        {
            System.out.print("FALLIDO: No se encontró una pelicula con el nombre especificado.");
        }

        /////////////////////////////
        dis.close();
        s1In.close();
        s1.close();
        dos.close();
        s1out.close();
	}
}

File: MovieClient.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <netdb.h> 

void error(const char *msg)
{
    perror(msg);
    exit(0);
}

int main(int argc, char *argv[])
{
    int sockfd, portno, n;
    struct sockaddr_in serv_addr;
    struct hostent *server;
    /////////////////// 
    char nombrePelicula[40], mensajeProtocolo[40], opcionConsulta[40];
    char resultadoBusqueda[40];
    char buffer[256];
    ///////////////////
    char *comma, *position;
    char number[40], resultadoFinal[40], ch;
    ///////////////////
    if (argc < 3) {
       fprintf(stderr,"usage %s hostname port\n", argv[0]);
       exit(0);
    }
    portno = atoi(argv[2]);
    sockfd = socket(AF_INET, SOCK_STREAM, 0);
    if (sockfd < 0) 
        error("ERROR opening socket");
    server = gethostbyname(argv[1]);
    if (server == NULL) {
        fprintf(stderr,"ERROR, no such host\n");
        exit(0);
    }
    bzero((char *) &serv_addr, sizeof(serv_addr));
    serv_addr.sin_family = AF_INET;
    bcopy((char *)server->h_addr, 
         (char *)&serv_addr.sin_addr.s_addr,
         server->h_length);
    serv_addr.sin_port = htons(portno);
    if (connect(sockfd,(struct sockaddr *) &serv_addr,sizeof(serv_addr)) < 0) 
        error("ERROR connecting");

    //////////////////////////////////////////////

    printf("Nombre de la pelicula a buscar: ");
    scanf(" %s", nombrePelicula);
    printf("\nOpciones de busqueda:\n- Genre            1\n- Lead Studio      2\n- Audience score   3\n- Profitability    4\n- Rotten Tomatoes  5\n- Worldwide Gross  6\n- Year             7\nElija la opcion: ");
    scanf(" %s", opcionConsulta);

    strcat(mensajeProtocolo, opcionConsulta);
    strcat(mensajeProtocolo, ",");
    strcat(mensajeProtocolo, nombrePelicula);

    // ENVIO DE DATOS AL SV
    bzero(buffer,256);
    strcpy(buffer, mensajeProtocolo);
    n = write(sockfd,buffer,strlen(buffer));
    if (n < 0) 
         error("ERROR writing to socket");
    // Server recibe el mensaje y lo procesa, para luego mandar el resultado:
    ////////////////////////////////////////////////

    bzero(buffer,256);
    n = read(sockfd,buffer,255);
    if (n < 0) error("ERROR reading from socket");
    strcpy(resultadoBusqueda,buffer);

    ///////////////////////////////////////////////

    comma = strchr(resultadoBusqueda, ',');
    position = resultadoBusqueda;
    while (comma)
    {
        int i = 0;

        while(position < comma)
        {
            number[i] = *position;
            i++;
            position++;
        }
        number[i]= '\0';
        strcpy(opcionConsulta, number);
        position++;
        comma = strchr (position, ',');
    }
    strcpy(resultadoFinal, position);

    ///////////////////////////////////////////////
    printf("El resultado de la busqueda fue: \n");
    if(atoi(opcionConsulta)==1)
    {
        printf("EXITOSO: ");
        printf("%s\n", resultadoFinal);
    }
    else
    {
        printf("FALLIDO: No se encontró una pelicula con el nombre especificado.\n");
    }
    ///////////////////////////////////////////////

    close(sockfd);

    return 0;
}